datastore/client/datastore_client.go

```diff
@@ -90,6 +93,52 @@ func (client DatastoreClient) Upsert(key string, value string) error {
 	return nil
 }
 
-func (client DatastoreClient) getUrlForKey(key string) string {
+func (client *DatastoreClient) getUrlForKey(key string) string {
 	return fmt.Sprintf("http://%v:%v/%v/%v", client.ipAddr, client.port, keyEndpoint, key)
 }
+
+/*
+Wait for healthy response
+*/
+func (client *DatastoreClient) WaitForHealthy(retries uint32, retriesDelayMilliseconds int) error {
+
+	var(
+		url = fmt.Sprintf("http://%v:%v/%v", client.ipAddr, client.port, healthcheckUrlSlug)
+		resp *http.Response
+		err error
+	)
+
+	for i := uint32(0); i < retries; i++ {
+		resp, err = client.makeHttpGetRequest(url)
+		if err == nil  {
+			break
+		}
+		time.Sleep(time.Duration(retriesDelayMilliseconds) * time.Millisecond)
+	}
+
+	body := resp.Body
+	defer body.Close()
+
+	bodyBytes, err := ioutil.ReadAll(body)
+	if err != nil {
+		return stacktrace.NewError("An error occurred reading the response body: %v", err)
```

Comment:
Same comment as before - this should use `stacktrace.Propagate`, not `NewError`, and shouldn't include the `err` text.

`return stacktrace.Propagate(err, "An error occurred reading the response body")` would be sufficient